server/util/tokenBalanceCache.ts

```diff
@@ -75,48 +76,47 @@ export default class TokenBalanceCache extends JobRunner<CacheT> {
       }));
 
     try {
-      const tokensFromListsResponses = await models.Token.findAll();
-      const tokensFromLists: TokenForumMeta[] = tokensFromListsResponses
-        .map((o) => {
-          return {
-            id: slugify(o.name),
-            address: o.address,
-            name: o.name,
-            symbol: o.symbol,
-            iconUrl: o.logoURI,
-          };
-        });
-
-      return [...tokens, ...tokensFromLists];
+      return tokens.concat(
+        (await this.models.Token.findAll()
+          .map((o) => {
+            return {
+              id: slugify(o.name),
+              address: o.address,
+              name: o.name,
+              symbol: o.symbol,
+              iconUrl: o.logoURI,
+            };
+          }))
+      );
     } catch (e) {
       log.error('An error occurred trying to access token lists', e.message);
     }
 
     return tokens;
   }
 
-  public getToken(searchAddress: string): TokenForumMeta {
-    return this._contracts.find(({ address }) => address === searchAddress);
+  public getToken(searchAddress: string): Promise<TokenForumMeta> {
+    return this.getTokens().then((o) => o.find(({ address }) => { return address.toLowerCase() === searchAddress.toLowerCase(); }));
```

Comment:
This is not what I had in mind -- you should be passing the address in as a DB query, not loading all tokens into memory and then searching them.